pkg/workloads/cortex/lib/context.py

```diff
@@ -168,11 +168,26 @@ def get_predictor_impl(self, api_name, project_dir):
             refresh_logger()
 
         try:
-            _validate_impl(impl, PREDICTOR_IMPL_VALIDATION)
+            classes = inspect.getmembers(impl, inspect.isclass)
+            if len(classes) == 0:
+                raise UserException("no class definitions found")
+            predictor_class = None
+            for class_df in classes:
+                if class_df[0] == "Predictor":
+                    if predictor_class is not None:
+                        raise UserException(
+                            "multiple definitions for Predictor class found (%s); please check your imports and class definitions and ensure that there is only one Predictor class definition",
+                            str(classes),
+                        )
+                    predictor_class = class_df[1]
+
+            if predictor_class is None:
+                raise UserException("definition for Predictor class not found")
```

Comment:
Can you incorporate `api["predictor"]["path"]` in any of the new errors you added here? e.g. `Predictor class not defined in {}`